### ISSUE TITLE
load_schema() does not recursively load referenced schemas.

### ISSUE THREAD

**Issue by adempus** [fastavro/fastavro#581]
Having an issue where the `load_schemas()` function doesn't recursively load referenced schemas assuming they can be found in files in the same directory and named with the convention <full_name>.avsc, as stated in the documentation. 

I replicated the file structure from the documentation like so:

<img width="220" alt="Screen Shot 2022-02-18 at 5 43 21 PM" src="https://user-images.githubusercontent.com/6502718/154771079-a0bbb311-7393-4413-94af-e8ef1417b5a9.png">

and ran the following code: 

```python
def test_avro_schema():
    parsed_schema = load_schema("namespace/Parent.avsc")
    print(f"parsed schema: \n{parsed_schema}")
```

got this error:

`fastavro._schema_common.UnknownType: namespace.Child`

Also, tried function `load_schema_ordered()`, but loading a list of schemas in a directory programmatically, is not guaranteed to load them in order by dependencies, unless in alphabetical order. Any workarounds?

**Comment by scottbelden** [comment]
It's difficult to know for sure what the problem is unless you are able to post the schemas here, but I think the problem is that the file is `Child.avsc` but the schema has a namespace. Therefore the file should be `namespace.Child.avsc`.

**Comment by adempus** [comment]
> It's difficult to know for sure what the problem is unless you are able to post the schemas here, but I think the problem is that the file is `Child.avsc` but the schema has a namespace. Therefore the file should be `namespace.Child.avsc`.

Problem is the child schema isn't detected even though both are in the same directory. The schemas are the same example show in the documentation https://fastavro.readthedocs.io/en/latest/schema.html#fastavro._schema_py.load_schema: 

```json
{
    "type": "record",
    "name": "Parent",
    "namespace": "namespace",
    "fields": [
        {
            "name": "child",
            "type": "Child"
        }
    ]
}
```
```json
{
    "type": "record",
    "namespace": "namespace",
    "name": "Child",
    "fields": []
}
```

**Comment by scottbelden** [comment]
And is the child schema file named `namespace.Child.avsc`? Based on the initial picture it looks like it is only named `Child.avsc`

**Comment by adempus** [comment]
> And is the child schema file named `namespace.Child.avsc`? Based on the initial picture it looks like it is only named `Child.avsc`

No, but I've tried that, still didn't work. Doesn't look like that convention is used in the documentation example. We're you able to get it working this way?

**Comment by scottbelden** [comment]
Yes, it does work for me:

```
$ ls foo
Parent.avsc		namespace.Child.avsc
```

```
$ cat foo/Parent.avsc
{
    "type": "record",
    "name": "Parent",
    "namespace": "namespace",
    "fields": [
        {
            "name": "child",
            "type": "Child"
        }
    ]
}
```

```
$ cat foo/namespace.Child.avsc
{
    "type": "record",
    "namespace": "namespace",
    "name": "Child",
    "fields": []
}
```

```
$ python -c "from fastavro.schema import load_schema; print(load_schema('foo/Parent.avsc'))"
{'type': 'record', 'name': 'namespace.Parent', 'fields': [{'name': 'child', 'type': {'type': 'record', 'name': 'namespace.Child', 'fields': []}}], '__fastavro_parsed': True, '__named_schemas': {'namespace.Child': {'type': 'record', 'name': 'namespace.Child', 'fields': []}, 'namespace.Parent': {'type': 'record', 'name': 'namespace.Parent', 'fields': [{'name': 'child', 'type': {'type': 'record', 'name': 'namespace.Child', 'fields': []}}]}}}
```

And I believe the [documentation](https://fastavro.readthedocs.io/en/latest/schema.html#fastavro._schema_py.load_schema) mentions `<full_name>.avsc`. You're right that the documentation uses `Parent.avsc` instead of `namespace.Parent.avsc` so I will fix that, but for the top most schema it actually doesn't matter because we don't check that the file name matches the contents (but I will still update the docs so it is consistent).

**Comment by scottbelden** [comment]
I'm going to close this as I believe it should be working as intended. If it is not then please re-open or open a new issue.
